### PR TITLE
handle background state better for coin flips CORE-10274

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1. If the app is backgrounded and it is in the middle of coin flips games, then attempt to run in the background.
2. If a coin flip message shows up for a non-active game, and the mobile app is not in the foreground, then ignore it.

### CODE REVIEW

**Comment by mmaxim** [comment]
Should go in before any release happens.

File: go/chat/flipmanager.go [hunk 1]
```diff
@@ -679,6 +692,56 @@ func (m *FlipManager) StartFlip(ctx context.Context, uid gregor1.UID, hostConvID
 	return m.dealer.StartFlipWithGameID(ctx, start, conv.GetConvID(), gameID)
 }
 
+func (m *FlipManager) shouldIgnoreInject(ctx context.Context, convID chat1.ConversationID,
+	gameID chat1.FlipGameID) bool {
+	if m.dealer.IsGameActive(ctx, convID, gameID) {
+		return false
+	}
+	// Ignore any flip messages for non-active games when not in the foreground
```
Comment (maxtaco) [inline go/chat/flipmanager.go]: update this comment....

Comment (mmaxim) [reply go/chat/flipmanager.go]: It applies to the following line

**Comment by mmaxim** [comment]
Testing locally

**Pull request merged**